doc/en/user/source/rest/api/stylegenerator.rst

```diff
@@ -0,0 +1,125 @@
+.. _rest_api_stylegenerator:
+
+Style Generator
+===============
+
+The Style Generator endpoint is used for generating default styles. Styles generated using this endpoint may be added to geoserver using the :ref:`styles <rest_api_styles>` endpoint.
+
+``/stylegenerator[.<format>]``
+------------------------------
```

Comment:
I like it - similar to recalculating a layer bounds.

I would recommend:

* ``/styles/foo.sld?generate=point`` to generate an point SLD style
* ``/styles/bar.css?generate=line`` to generate an line string css style

And then PUT/POST can just be used based on if a new style is being created or not.